AsyncDisplayKit/ASMapNode.mm

```diff
@@ -279,7 +279,7 @@ - (void)setAnnotations:(NSArray *)annotations
 #pragma mark - Layout
 - (void)setSnapshotSizeWithReloadIfNeeded:(CGSize)snapshotSize
 {
-  if (!CGSizeEqualToSize(self.options.size, snapshotSize)) {
+  if (snapshotSize.height > 0 && snapshotSize.width > 0 && !CGSizeEqualToSize(self.options.size, snapshotSize)) {
```

Comment:
Is there any possibility that a new snapshot could be triggered using the old size, after the point at which we attempted to set the snapshot size to zero but then avoided entering this condition?

It seems like we may need some kind of flag that invalidates the current size. Maybe a safer fallback is to unconditionally set the size of the options, but guard against calling into the snapshoter with that options if the size is equal to the zero size.